ateam_ai/src/ateam_ai_node.cpp

```diff
@@ -163,6 +170,57 @@ class ATeamAINode : public rclcpp::Node
     ball_state.vel.y() = ball_state_msg->twist.linear.y;
   }
 
+  void field_callback(const ssl_league_msgs::msg::VisionGeometryFieldSize::SharedPtr field_msg)
+  {
+
+    Field field {
+        .field_length   = field_msg->field_length,
+        .field_width    = field_msg->field_width,
+        .goal_width     = field_msg->goal_width,
+        .goal_depth     = field_msg->goal_depth,
+        .boundary_width = field_msg->boundary_width
+    };
+
+    auto check_field_line_name = [](ssl_league_msgs::msg::VisionFieldLineSegment line_msg, std::string target_name) -> bool {
+        return line_msg.name == target_name; 
+    };
+    std::array<std::string, 4> field_bound_names = {"TopTouchLine", "BottomTouchLine", "LeftGoalLine", "RightGoalLine"};
+    for (size_t i = 0; i < field_bound_names.size(); i++) {
+        auto& name = field_bound_names.at(i);
+        auto itr = std::find_if(begin(field_msg->field_lines), end(field_msg->field_lines), std::bind(check_field_line_name, std::placeholders::_1, name));
```

Comment:
I'm on the edge if this would be better, but if you pull in the lambda into the for loop, you can capture the name reference instead of having to bind it. And if you swap to auto, it could even be inline?

```
auto itr = std::find_if(begin(field_msg->field_lines), end(field_msg->field_lines), [&name](auto msg) { return msg.name == name; });
```

It honestly works as is and will never need to be changed so it doesn't really matter, so it's fine to leave